bmsSender.c

```diff
@@ -0,0 +1,55 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "bmsSender.h"
+
+void generateBatteryData(batteryParameters *ptr_BatteryParam)
+{
+	//generate data for Battery parameters
+	for(int index = 0; index < NUM_OF_READINGS; index++)
+    {
+		ptr_BatteryParam->temperature[index] = (rand()%(TEMP_MAX - TEMP_MIN + 1)) + TEMP_MIN;
+		ptr_BatteryParam->soc[index] = (rand()%(SOC_MAX - SOC_MIN + 1)) + SOC_MIN;
+        printBatteryData(index + 1, ptr_BatteryParam->temperature[index], ptr_BatteryParam->soc[index]);
+        writeToCsv(index + 1, ptr_BatteryParam->temperature[index], ptr_BatteryParam->soc[index]);
```

Comment:
SRP violated. These functions can be called in status_en senderProcess(void) instead.